dispatcher/dispatcher.py

```diff
@@ -0,0 +1,64 @@
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations
+
+import logging
+import types
+from typing import TYPE_CHECKING, Any, Generic, List, Mapping
+
+if TYPE_CHECKING:
+    from ._types import BotT, CoroFunc
+
+__all__ = ("Dispatcher",)
+
+_log = logging.getLogger(__name__)
+
+
+class Dispatcher(Generic[BotT]):
+    """
+    A class to handles custom event dispatching.
+
+    .. versionadded:: 0.0.1
+
+    Parameters
+    ----------
+    bot: BotT
+        The bot instance to work with.
+    """
+
+    bot: BotT
```

Comment:
Is this required to be here?

Reply:
Uh yes. Isn't it?

Reply:
you're already specifying it at line 31 so i don't think

Reply:
The class-level annotation is need for type-checker to know that the `bot` attr on Dispatcher must be of type BotT. The annotation on line 31 is required for type-checker to know that Dispatcher accepts only values of type BotT for the bot parameter. Without the class-level annotation, type-checker would complain that member bot is not defined on type Dispatcher. Without the line 31 annotation type-checker would complain that value of type Any cannot be assigned to variable of type BotT.